tests/test_eth.py

```diff
@@ -92,6 +92,30 @@ def did_evm_execute_instruction_callback(self, state, instruction, arguments, re
         self.assertIn('STOP', context)
         self.assertIn('REVERT', context)
 
+    def test_graceful_handle_no_alive_states(self):
+        """
+        If there are no alive states, or no initial states, we should not crash. issue #795
+        """
+        # initiate the blockchain
+        m = ManticoreEVM()
+        source_code = '''
+        contract Simple {
+            function f(uint a) payable public {
+                if (a == 65) {
+                    revert();
+                }
+            }
+        }
+        '''
+
+        # Initiate the accounts
+        user_account = m.create_account(balance=1000)
+        contract_account = m.solidity_create_contract(source_code, owner=user_account, balance=0)
+
+        contract_account.f(1)  # it works
+        contract_account.f(65)  # it works
+        contract_account.f(m.SValue)  # no alive states, but try to run a tx anyway
```

Comment:
I'd expect this to raise an exception. No?

Reply:
Hm, so i originally was thinking that it makes sense for ManticoreEVM.run() to simply return if there are no states to analyze. The run() function executes the various alive states. If there are no alive states, there is nothing to do, so just return!

However, the question of whether or not executing a transaction when there are no states is a higher level question, and a different one. I think you're right; trying to execute a transaction when there are no states should raise an Exception. it could be confusing for a user, if they try to execute transactions and it appears to 'work', but in reality manticore is doing nothing.

Reply:
I still like keeping the behavior of .run() mentioned above, but I will include a check at a different layer to implement the exception.

Reply:
So running over no states is not an exception.